dev/tasks/homebrew-formulae/github.macos.yml

```diff
@@ -29,6 +29,10 @@ jobs:
       {{ macros.configure_homebrew_arrow(formula)|indent }}
       - name: Test formula
         run: |
+          # The github runners have a non-brew 3.11 installed and the existing symlinks cause
+          # the installation of brew 3.11 to fail
+          find /usr/local/bin -lname '*/Library/Frameworks/Python.framework/*' -delete
```

Comment:
IMHO, this is a pretty expansive glob that could impact a number of different files — any way that we could make it be a bit more specific to the issue we're seeing? Either (or both of): looking for 3.11 specifically, only running this on (specific) github runners? Or maybe even better: `brew link --overwrite python@3.11` which is suggested in the message?

Reply:
Yeah the idea was to not have to add multiple things manually after retrying a bunch but being eplicit is probably for the best (e.g. `rm '/usr/local/bin/2to3-3.11'`). 

This is only done on the affected runners (non-10.13), using `overwrite` would require us to change it in the brew internals which would affect everyone not just these (wrongly configured) CI runners.

I'll switch this to the explicit removal, that will probably also fix the brew job for which the glob apparently failed...